src/app/clusters/operational-state-server/operational-state-delegate.h

```diff
@@ -0,0 +1,393 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/util/af-enums.h>
+#include <lib/support/CommonIterator.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+constexpr size_t kOperationalStateLabelMaxSize   = 64u;
+constexpr size_t kOperationalErrorLabelMaxSize   = 64u;
+constexpr size_t kOperationalErrorDetailsMaxSize = 64u;
+constexpr size_t kOperationalPhaseNameMaxSize    = 64u;
+
+/**
+ * A class wrap the operational state of operational state cluster
+ */
+struct GenericOperationalState : public chip::app::Clusters::detail::Structs::OperationalStateStruct::Type
+{
+    GenericOperationalState(uint8_t state, const char * label = nullptr, size_t labelLen = 0) { set(state, label, labelLen); }
+
+    GenericOperationalState(const GenericOperationalState & op)
+    {
+        set(op.operationalStateID, op.operationalStateLabel.HasValue() ? op.OperationalStateLabel : nullptr,
+            op.operationalStateLabel.HasValue() ? sizeof(op.OperationalStateLabel) : 0);
+    }
+
+    GenericOperationalState & operator=(const GenericOperationalState & op)
+    {
+        set(op.operationalStateID, op.operationalStateLabel.HasValue() ? op.OperationalStateLabel : nullptr,
+            op.operationalStateLabel.HasValue() ? sizeof(op.OperationalStateLabel) : 0);
+
+        return *this;
+    }
+
+    void set(uint8_t state, const char * label = nullptr, size_t labelLen = 0)
+    {
+        operationalStateID = state;
+        if (label == nullptr)
+        {
+            operationalStateLabel = Optional<chip::CharSpan>::Missing();
```

Comment:
```suggestion
            operationalStateLabel = NullOptional;
```
or just `operationalStateLabel.ClearValue()`.

Reply:
Commit suggestion.